canonical/maas-ui
# feat(machines): collapse groups via the API

## Done

- Update the machine list to toggle collapsed groups via the API.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the machine list.
- On one of the group headers click the "-" button.
- The machine list should reload and the group should be collapsed.
- Refresh the page and the machine list should load with the group collapsed.

## Fixes

Fixes: canonical/app-tribe#1123.
